lib/SILGen/SILGenRequests.cpp

```diff
@@ -46,14 +46,18 @@ SourceLoc swift::extractNearestSourceLoc(const SILGenDescriptor &desc) {
   return SourceLoc();
 }
 
-evaluator::DependencySource SILGenSourceFileRequest::readDependencySource(
+evaluator::DependencySource SILGenerationRequest::readDependencySource(
     const evaluator::DependencyCollector &e) const {
   auto &desc = std::get<0>(getStorage());
+
+  // Whole module doesn't affect the dependency source.
+  if (auto *mod = desc.context.dyn_cast<ModuleDecl *>()) {
+    return {e.getActiveDependencySourceOrNull(), e.getActiveSourceScope()};
```

Comment:
You can also explicitly return `{nullptr, active-scope}`, but I suppose this is more conservatively correct.

Reply:
`nullptr` probably makes it a little more clear, I'll switch over to that for now.